### DISCUSSION TITLE
Question about training set, and training config. Does size of the training images have to be the same for all files? Will it improve results if it will the same for training and inference? Does YOLO5 automatically use image augmentation while training?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [ultralytics/yolov5#9919] [Q&A]
Question about training set, and training config. Does size of the training images have to be the same for all files? Does YOLO5 automatically use image augmentation while training?

python train.py --img 640 --batch 16 --epochs 3 --data coco128.yaml --weights yolov5s.pt

Does it mean that it is better to have all images in 640 by 640 in my training set?
Or does it only considers the width of an image? 
Is it better to set it to --img 720 if I am going to use videos in 720p resolution? Will it improve accuracy when prediction objects? 
What happens if I have an image in my training set with a completely different resolution? Won't it hurt performance or the image will be resized to --img 640? 
Does it drastically improve training time and memory consumption? 
Is it advisable to increase image resolution (especially for detecting small objects?)

Btw Is it better to start training on preexisting weights? 

Does YOLO5 automatically apply image augmentation? How do I add additional augmentations or change applied? 
What additional image augmentations are recommended?  
Image rotation, grayscale, image resizing?

**Comment by glenn-jocher** [comment]
Training images can be any size, as they are all resized to --imgsz during training, i.e.
```
python train.py --imgsz 640
```

**Reply by martinenkoEduard** [reply]
Are they resized to a square image? Or is it only height of the image?

What about other questions? ))))

**Reply by glenn-jocher** [reply]
@martinenkoEduard training images are square mosaics at imgsz x imgsz

You should always start training from pretrained weights for best results, i.e.
```
python train.py --weights yolov5s
```
